drivers/include/mtd_flashpage.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2018 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_mtd_flashpage   Flashpage MTD
+ * @ingroup     drivers_storage
+ * @brief       Driver for internal flash devices implementing flashpage interface
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the flashpage memory driver
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ */
+
+#ifndef MTD_FLASHPAGE_H
+#define MTD_FLASHPAGE_H
+
+#include "mtd.h"
+#include "periph/flashpage.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define MTD_FLASHPAGE_INIT_VAL(_pages_per_sector) { \
+    .driver = &mtd_flashpage_driver, \
+    .sector_count = FLASHPAGE_NUMOF, \
```

Comment:
So by default, IIUC the MTD can take the whole flash. Can we make this configurable at build time with a more reasonable default value (half of the flash) ?

Reply:
@vincent-d, do you think it would make sense to limit the default size used to half of the flash size ?

Reply:
I took a new look to this and there is 2 options:
- we leave it as-is and rely on the configuration of the filesystem for the offset in the mtd (i.e. the mtd allows access to the whole internal flash and the filesystem is configured correctly to not write outside its area)
- we change the mtd_flashpage to let the filesystem access only a part of the internal flash, but we need to have a specific `mtd_flashpage_t` descriptor to store the offset

If we only change `sector_count` the filesystem will be able to access the number of sector passed through this field, but it will always start from address 0 (the first sector)

Reply:
I'm not sure to understand everything clearly.

If the first option is kept, we'll have to trust the filesystem implementation and most of them come from external packages. My question in this case is how do you configure this external filesystem ? (I checked littlefs and I'm not sure to understand how to do that)

If the second option is kept, could the offset be automatically determined from the number of sectors defined (using offset =  flashsize - number_of_sector) ?

> he filesystem will be able to access the number of sector passed through this field, but it will always start from address 0 

Is this an issue ?